pages/docs/tutorials/native/interop-with-c.md

```diff
@@ -0,0 +1,140 @@
+need to relative to the folder where the definition file is, or by a available on the system path (in our case it would be `/usr/bin/curl`).
+
+The second line is the `headerFilter`. This is used to denote what exactly we want included. In C, when one file references another file with the `#include` directive, 
+all the headers are also included. Sometimes this may not be needed, and we can use this parameter, [using glob patterns](https://en.wikipedia.org/wiki/Glob_(programming)), to fine tune things. 
+
+The third and forth line are about providing linker options, which can vary based on different target platforms. In our cause we are defining it for macOS and Linux. 
+Compiler options can also be specific, albeit in this case it is not necessary. 
+
+The convention that is followed is that each library gets its own definition file, usually named the same as the library. For more information on all
+the options available to `cinterop`, see [README available on GitHub](https://github.com/JetBrains/kotlin-native/blob/master/INTEROP.md)
+
+Once we have the definition file ready, we can invoke `cinterop` and have it generate the bindings for us
+
+    cinterop -def libcurl.def -o build/c_interop/libcurl
+    
+The output should be a Kotlin compiled library (extension `.klib`) and the Kotlin source header files. We're also instructing `cinterop`
+to output the contents to the folder `build/c_interop/libcurl` which we'll later use as input by `konanc`. It's important this folder is created
+beforehand as `cinterop` needs an existing folder. The *c_interop* is a convention that is followed for libraries that are for interop.
+
+
+
+## Consuming the Kotlin API
+
+Now that we have our library and Kotlin stubs, we can consume them from our application. To keep things simple, in this tutorial we're going to convert one of the simplest 
+libcurl examples over to Kotlin. 
+
+The code in question is from the [simple](https://curl.haxx.se/libcurl/c/simple.html) example (comments removed for brevity)
+
+```c
+#include <stdio.h>
+#include <curl/curl.h>
+ 
+int main(void)
+{
+  CURL *curl;
+  CURLcode res;
+ 
+  curl = curl_easy_init();
+  if(curl) {
+    curl_easy_setopt(curl, CURLOPT_URL, "http://example.com");
+    curl_easy_setopt(curl, CURLOPT_FOLLOWLOCATION, 1L);
+ 
+    res = curl_easy_perform(curl);
+    if(res != CURLE_OK)
+      fprintf(stderr, "curl_easy_perform() failed: %s\n",
+              curl_easy_strerror(res));
+    curl_easy_cleanup(curl);
+  }
+  return 0;
+}
+```
+
+The first thing we'll need is a Kotlin file called `Main.kt` with the `main` function defined in it and then proceed to translate each line
+
+```kotlin
+import libcurl.*
+
+fun main(args: Array<String>) {
+    val curl = curl_easy_init()
+    if (curl != null) {
+        curl_easy_setopt(curl, CURLOPT_URL, "http://example.com")
+        curl_easy_setopt(curl, CURLOPT_FOLLOWLOCATION, 1L)
+    }
```

Comment:
if `curl != null` is false, you need to skip the entire body, not just the setopt calls.